File: 2-Patron/python/MonstruoComun.py
from typing import Hashable
import Criatura
class MonstruoComun(Criatura):
    def __init__(self, nombre, defensa, ataque, nivel, habilidad, ultimate, resistencia):
        super().__init__(nombre, defensa, ataque, nivel)
        self.__habilidad = habilidad
        self.__ultimate = ultimate
        self.__resistencia = resistencia

    @property
    def habilidad(self):
        return self.__habilidad

    @habilidad.setter
    def habilidad(self, new):
        self.__habilidad = new

    @property
    def ultimate(self):
        return self.__ultimate

    @ultimate.setter
    def ultimate(self, new):
        self.__ultimate = new

    @property
    def resistencia(self):
        return self.__resistencia

    @resistencia.setter
    def resistencia(self, new):
        self.__resistencia = new

    def reducirResistencia(self):
        self.__resistencia -= 1

    def decSalud(self, decremento):
        self.__salud -= decremento // self.__
    
    def incNivel(self):
        self.__nivel += 2

    def verStats(self):
        return f"{super().verStats()}\n\
                 Habilidad: {self.__sígueme}\n \
                 Ultimate: {self.__ultimate}\n \
                 Resistencia: {self.__resistencia}\n"

File: 2-Patron/php/MonstruoComun.php
<?php
class MostruoComun extends Criatura
{
  private $habilidad;
  private $ultimate;
  private $resistencia;

  function __construct($nombre, $defensa, $ataque, $nivel, $habilidad, $ultimate, $resistencia)
  {
    parent::__construct($nombre, $defensa, $ataque, $nivel);
    $this->habilidad = $habilidad;
    $this->ultimate = $ultimate;
    $this->resistencia = $resistencia;
  }

  function getHabilidad()
  {
    return $this->habilidad;
  }

  function setHabilidad($nHabilidad)
  {
    $this->habilidad = $nHabilidad;
  }

  function getUltimate()
  {
    return $this->ultimate;
  }

  function setUltimate($nUltimate)
  {
    $this->ultimate = $nUltimate;
  }

  function getResistencia()
  {
    return $this->resistencia;
  }

  function setResistencia($nResistencia)
  {
    $this->resistencia = $nResistencia;
  }

  function decSalud($decremento)
  {
    $this->salud -= $decremento;
  }

  function incNivel()
  {
    $this->nivel += 2;
  }

  function verStats()
  {
    return  parent::verStats() .
            "Habilidad: " . $this->arma .
            "Ultimate: " . $this->ultimate .
            "Experiencia" $this->experiencia;
  }
}
?>
